pkg/proxy/iptables/proxier.go

```diff
@@ -572,45 +590,56 @@ func (proxier *Proxier) OnEndpointsSynced() {
 	proxier.syncProxyRules(syncReasonEndpoints)
 }
 
-// Convert a slice of api.Endpoints objects into a map of service-port -> endpoints.
-func buildNewEndpointsMap(allEndpoints endpointsMap, curMap proxyEndpointMap, hostname string) (newMap proxyEndpointMap, hcEndpoints map[types.NamespacedName]int, staleSet map[endpointServicePair]bool) {
+func (proxier *Proxier) mergeEndpointsMap(endpointsMap proxyEndpointsMap) {
```

Comment:
For targetted test coverage, I would make this a non-method function taking two maps, returning a mutated form of the first.  Then a simple unit test.

Even better (maybe) just define `func (epmap endpointsMap) merge(other endpontsMap)` and `unmerge(...)` ?